sdks/node/src/wandb/settings.ts

```diff
@@ -1,45 +1,78 @@
-import {defaultDomain, defaultHost, getUrls} from '../urls';
+import {defaultHost, getUrls} from '../urls';
 import {Netrc} from '../utils/netrc';
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import * as os from 'node:os';
+import * as ini from 'ini';
 
-export function getApiKey(host: string): string {
-  let apiKey = process.env.WANDB_API_KEY;
-  if (!apiKey) {
+function getBaseUrlFromConfig(): string | undefined {
+  // Check environment first
+  const envBaseUrl = process.env.WANDB_BASE_URL;
+  if (envBaseUrl) {
+    return envBaseUrl;
+  }
+
+  // Try to read from config file
+  try {
+    const configPath = path.join(os.homedir(), '.config', 'wandb', 'settings');
+    if (fs.existsSync(configPath)) {
+      const config = ini.parse(fs.readFileSync(configPath, 'utf-8'));
+      if (config.default?.wandb_base_url) {
+        return config.default.wandb_base_url;
+      }
+    }
+  } catch (error) {
+    // Silently fail if we can't read the config file
+    // This could happen in restricted environments like Deno
```

Comment:
Why fail silently?

Reply:
The failure case is if the file exists but we're unable to parse it.  This should be exceedingly rare, but since we can't really do anything about it from this SDK (wandb login --host=xxxx sets it), I figure we just pretend it doesn't exist and default to api.wandb.ai